operators/src/processing/temporal_raster_aggregation/mean_aggregation_subquery.rs

```diff
@@ -0,0 +1,227 @@
+use futures::{Future, FutureExt, TryFuture};
+use geoengine_datatypes::{
+    primitives::{SpatialBounded, TimeInstance, TimeInterval, TimeStep},
+    raster::{
+        EmptyGrid2D, GeoTransform, Grid2D, GridIdx2D, GridOrEmpty, GridOrEmpty2D, GridShapeAccess,
+        NoDataValue, Pixel, RasterTile2D, TileInformation,
+    },
+};
+use num_traits::AsPrimitive;
+
+use crate::{
+    adapters::{FoldTileAccu, SubQueryTileAggregator},
+    engine::QueryRectangle,
+    util::Result,
+};
+
+pub fn mean_tile_fold_future<T>(
+    accu: TemporalMeanTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<TemporalMeanTileAccu<T>>>
+where
+    T: Pixel,
+{
+    tokio::task::spawn_blocking(|| {
+        let mut accu = accu;
+        accu.add_tile(tile)?;
+        Ok(accu)
+    })
+    .then(async move |x| match x {
+        Ok(r) => r,
+        Err(e) => Err(e.into()),
+    })
+}
+
+#[derive(Debug, Clone)]
+pub struct TemporalMeanTileAccu<T> {
+    time: TimeInterval,
+    tile_position: GridIdx2D,
+    global_geo_transform: GeoTransform,
+
+    value_grid: GridOrEmpty2D<f64>,
+    count_grid: Grid2D<u64>,
+    ignore_no_data: bool,
+    out_no_data_value: T,
+}
+
+impl<T> TemporalMeanTileAccu<T> {
+    pub fn add_tile(&mut self, in_tile: RasterTile2D<T>) -> Result<()>
+    where
+        T: Copy + AsPrimitive<f64> + Pixel,
+    {
+        self.time = self.time.union(&in_tile.time)?;
+
+        let in_tile_grid = match in_tile.grid_array {
+            GridOrEmpty::Grid(g) => g,
+            GridOrEmpty::Empty(_) => return Ok(()),
```

Comment:
Isn't Empty like adding NODATA?

Reply:
depends if it is the initial case

Reply:
we need the         initial_state: false, logic i guess

Reply:
5708885